src/select/dropdown.js

```diff
@@ -95,8 +83,26 @@ export default function SelectDropDown(props: DropDownPropsT) {
               );
             },
           },
+          component: (optionProps: OptionListPropsT) => {
```

Comment:
This is fine for now but can we sync up about why such custom overrides are needed here. What behavior is this `DropDown` trying to capture? Are we just trying not to change the external API here?

Reply:
еthis is the way to add click handler not to inner custom label, but to the full body of menu item. I couldn't provide it via static props as they are same for all items, but we have different logic per each item

Reply:
But, that's probably a separate conversation not in the scope of this PR. I remember looking into this before and couldn't figure out how to do it without changing the API for Select overrides.

Reply:
`getRequiredItemProps` doesn't work as it is defined in `menu/stateful-container.js` and the one I provide in props just doesn't come through. Cause I use stateful menu since I have to( we had discussion on that before), otherwise it doesn't change state properly.

And below doesn't work as well as I said props are the same for all Options, which I can't use, I need dynamic props for each of them
```<StatefulMenu
     overrides={{
        List: { /* this is where you're doing per-item override now */ },
        Option: { /* this should be where you're doing per-item override */},
     }}
/>```

Reply:
Yeah I realized `getRequiredItemProps` wouldn't work in this case. Maybe the fix there is to merge user's `getRequiredItemProps` with the internal `getRequiredItemProps` in `StatefulContainer`.

But what I meant with the second option was to define your own `Option` component that would take in the `item` as a prop, and you can encapsulate all your logic in there. That's what the API is meant for (which seems like what you're doing here through using all the other APIs separately).

Reply:
Second option is no difference for me than what I provided so far.

I think adding `getRequiredItemProps` from props to container seems reasonable so far and requires just one line of code

Reply:
I updated with the use of method from props